### PR TITLE
Hopkinsons law

### PR DESCRIPTION
**Pull request by R3hankhan123**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by prawineterna18** [comment]
Stop sending me this notifications
Please

File: electronics/hopkinson's_law.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    >>> hopkinsons_law(reluctance=0, magnetomotive_force=-10, current=1)
+    {'reluctance': -10.0}
+    >>> hopkinsons_law(magnetomotive_force=0, current=-1.5, reluctance=2)
+    {'magnetomotive_force': -3.0}
+    """
+    if (magnetomotive_force, current, reluctance).count(0) != 1:
+        raise ValueError("One and only one argument must be 0")
+    if reluctance < 0:
+        raise ValueError("Reluctance cannot be negative")
+    if magnetomotive_force == 0:
```
Comment (CaedenPH) [inline electronics/hopkinson's_law.py:27]:
```suggestion

    if magnetomotive_force == 0:
```

File: electronics/hopkinson's_law.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#https://en.wikipedia.org/wiki/Magnetic_circuit#Hopkinson's_law
+from __future__ import annotations
+def hopkinsons_law(magnetomotive_force:float, current:float, reluctance:float) -> dict[str, float]:
```
Comment (CaedenPH) [inline electronics/hopkinson's_law.py:3]:
```suggestion


def hopkinsons_law(magnetomotive_force:float, current:float, reluctance:float) -> dict[str, float]:
```

File: electronics/hopkinson's_law.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        return {"current": magnetomotive_force / reluctance}
+    elif reluctance == 0:
+        return {"reluctance": magnetomotive_force / current}
+    else:
+        raise ValueError("Exactly one argument must be 0")
+
+        
+if __name__ == "__main__":
+    import doctest
+    doctest.testmod()
```
Comment (CaedenPH) [inline electronics/hopkinson's_law.py:39]:
```suggestion

    doctest.testmod()        
    
```

**Review by CaedenPH — APPROVED** [review]

**Pull request closed without merging**